Makefile

```diff
@@ -5,32 +5,33 @@ CLI_DIST_DIR := $(BUILD_DIR)/cli_dist
 DIST_DIR := $(BUILD_DIR)/dist
 GIT_COMMIT := $(shell git rev-parse HEAD)
 
-S3_BUCKET := infinity-artifacts
-S3_PREFIX := autodelete7d
+S3_BUCKET ?= infinity-artifacts
+S3_PREFIX ?= autodelete7d
+SPARK_REPO_URL ?= https://github.com/mesosphere/spark
 
 .ONESHELL:
 SHELL := /bin/bash
 .SHELLFLAGS = -ec
 
 # This image can be used to build spark dist and run tests
-DOCKER_BUILD_IMAGE := mesosphere/spark-build:$(GIT_COMMIT)
+DOCKER_BUILD_IMAGE ?= mesosphere/spark-build:$(GIT_COMMIT)
 docker-build:
 	docker build -t $(DOCKER_BUILD_IMAGE) .
 	echo $(DOCKER_BUILD_IMAGE) > $@
 
 # Pulls the spark distribution listed in the manifest as default
-SPARK_DIST_URI := $(shell jq ".default_spark_dist.uri" "$(ROOT_DIR)/manifest.json")
+SPARK_DIST_URI ?= $(shell jq ".default_spark_dist.uri" "$(ROOT_DIR)/manifest.json")
 manifest-dist:
 	mkdir -p $(DIST_DIR)
 	pushd $(DIST_DIR)
 	wget $(SPARK_DIST_URI)
 	popd
 
-HADOOP_VERSION := $(shell jq ".default_spark_dist.hadoop_version" "$(ROOT_DIR)/manifest.json")
+HADOOP_VERSION ?= $(shell jq ".default_spark_dist.hadoop_version" "$(ROOT_DIR)/manifest.json")
 
-SPARK_DIR := $(ROOT_DIR)/spark
+SPARK_DIR ?= $(ROOT_DIR)/spark
```

Comment:
The good thing about this is that if I set: export SPARK_DIR=$(pwd)/../spark
from my cli then the target below will not be invoked (no cling) which allows me to
build from an existing folder.